taichi/ir/statements.h

```diff
@@ -1082,8 +1082,9 @@ class ReferenceStmt : public Stmt, public ir_traits::Load {
 class GetElementStmt : public Stmt {
  public:
   Stmt *src;
-  int index;
-  GetElementStmt(Stmt *src, int index) : src(src), index(index) {
+  std::vector<int> index;
+  GetElementStmt(Stmt *src, std::vector<int> index)
+      : src(src), index(std::move(index)) {
```

Comment:
```suggestion
      : src(src), index(index) {
```

Reply:
Either we move it from the caller (which is destructive, not sure that's the best idea)
Or we make a copy from a const reference

Reply:
If an lvalue is passed, it will first be copied and then the copied one is moved, so it's not destructive. IIUC If an rvalue is passed, if we pass by const reference, it is copied (or maybe it will be optimized out, I'm not sure). However if we pass by value, it is not copied. So  I think passing by value is better than by const reference, at least not worse than.

Reply:
If rvalue is passed and if it's const reference it won't be copied. In the end the result is the same. However a const reference is cleaner than passing in a copy then move because in the constructor access to `index` will then be invalid.